nixos/modules/services/mail/rss2email.nix

```diff
@@ -0,0 +1,226 @@
+      }
+      body {
+        font: 12pt Arial;
+      }
+      a:link {
+        font: 12pt Arial;
+        font-weight: bold;
+        color: #0000cc;
+      }
+      blockquote {
+        font-family: monospace;
+      }
+      .header {
+        background: #e0ecff;
+        border-bottom: solid 4px #c3d9ff;
+        padding: 5px;
+        margin-top: 0px;
+        color: red;
+      }
+      .header a {
+        font-size: 20px;
+        text-decoration: none;
+      }
+      .footer {
+        background: #c3d9ff;
+        border-top: solid 4px #c3d9ff;
+        padding: 5px;
+        margin-bottom: 0px;
+      }
+      #entry {
+        border: solid 4px #c3d9ff;
+      }
+      #body {
+        margin-left: 5px;
+        margin-right: 5px;
+      }
+    '';
+    unicode-snob = "False";
+    links-after-each-paragraph = "False";
+    body-width = "0";
+    email-protocol = "sendmail";
+    sendmail = "sendmail";
+    smtp-auth = "False";
+    smtp-username = "username";
+    smtp-password = "password";
+    smtp-server = "smtp.yourisp.net:25";
+    smtp-ssl = "False";
+    smtp-ssl-protocol = "SSLv3";
+    imap-auth = "False";
+    imap-username = "username";
+    imap-password = "password";
+    imap-server = "imap.yourisp.net";
+    imap-port = "143";
+    imap-ssl = "False";
+    imap-mailbox = "INBOX";
+    verbose = "warning";
+  };
+
+in {
+
+  ###### interface
+
+  options = {
+
+    services.rss2email = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable rss2email.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.rss2email;
+        defaultText = "pkgs.rss2email";
+        description = "The rss2email package to use.";
+      };
+
+      to = mkOption {
+        type = types.str;
+        description = "Mail address to which to send emails";
+      };
+
+      interval = mkOption {
+        type = types.str;
+        default = "1h";
+        description = "How often to check the feeds, in systemd interval format";
+      };
+
+      configuration = mkOption {
+        type = types.attrs;
+        description = ''
+          The configuration to give rss2email.
+
+          Default will use system-wide <literal>sendmail</literal> to send the
+          email. This is rss2email's default when running
+          <literal>r2e new</literal>.
+
+          This set contains key-value associations that will be set in the
+          <literal>[DEFAULT]</literal> block along with the
+          <literal>to</literal> parameter.
+        '';
+        example = lib.literalExample ''
+          {
+            from = "user@rss2email.invalid";
+
+            # You likely want to use something like this for css
+            css = builtins.replaceStrings ["\n"] [""] '''
+              body {
+                font: 15pt Arial;
+              }
+            ''';
+          }
+        '';
+        default = default-config;
```

Comment:
The problem with this is that setting *any* value in this option will remove all defaults. Also: is it really required to set these defaults in the NixOS config? If the program sets defaults itself if it can't find the values, then this isn't needed (also has the advantage that we don't need to update the defaults here if they change). If the defaults really do need to be set in order for it to work, I suggest doing this:
- Change the type to `attrsOf (either str (either int bool))`, this makes merging work as one would expect. When generating the result, make sure ints and bools get translated correctly (maybe the toINI generator does that already though).
- Set the defaults by doing `services.rss2email.config = mapAttrs (n: mkDefault) { use-8bit = false; ... };` in the `config` section of this module. This will make it such that all defaults aren't lost if a single gets changed. This also allows the user to see the value of the defaults by looking at the value of this option.

Reply:
I can confirm that the program uses its defaults internally without issues, thanks! Didn't think of this.

BTW, the merging-with-default behavior was handled before with `DEFAULT = default-config // cfg.configuration // { to = cfg.to; };`, which took `default-config`.

I'm still fixing the type as you advised so that the user can define parts of the option at different places. :)

I'm taking `default-config` and making it into a `literalExample`, so that the user has an idea of what they can use :)